aws/rust-runtime/aws-auth/src/provider/lazy_caching.rs

```diff
@@ -186,8 +202,9 @@ pub mod builder {
             );
             LazyCachingCredentialsProvider::new(
                 SystemTimeSource,
-                self.refresh.expect("refresh provider is required"),
-                self.refresh_timeout.unwrap_or(DEFAULT_REFRESH_TIMEOUT),
+                self.sleep.expect("sleep_impl is required"),
```

Comment:
if the tokio feature is enabled, we could consider defaulting to tokio-sleep here